app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -87,6 +92,29 @@ public RightMeshController(User user, MeshIMDao dao,
                 this.dao.updateUsers(user);
             }
         }).start();
+        // creating a thread that checks for undelivered packages periodically
+        //resend data every 8 seconds (allow time for multihop?)
+        Timer undeliveredPackageTimer = new Timer();
+        undeliveredPackageTimer.scheduleAtFixedRate(new TimerTask() {
```

Comment:
Does this thread get stopped when we shut the app down? Is there any way we can only have it run when there are messages in the queue to be sent? Like on a remove from the queue we can stop it, and when we insert to the queue if it's empty we can start it?

Reply:
since its running in a service class, I would assume it will run till the service runs. Start it and stoping it is a very neat idea, I will look into it.